flight/tasks/imu.py

```diff
@@ -33,31 +33,36 @@ def __init__(self, id):
         self.name = "IMU"
 
     async def main_task(self):
-
         if SM.current_state == STATES.NOMINAL:
 
             if not DH.data_process_exists("imu"):
                 DH.register_data_process(
                     "imu", "Lfffffffff", True, data_limit=100000, write_interval=5, circular_buffer_size=5
                 )
 
-            accel = SATELLITE.IMU.accel()
-            mag = SATELLITE.IMU.mag()
-            gyro = SATELLITE.IMU.gyro()
+            if SATELLITE.IMU_AVAILABLE:
+                if SATELLITE.IMU_NAME == "BNO08X":
+                    accel = SATELLITE.IMU.acceleration
+                    mag = SATELLITE.IMU.magnetic
+                    gyro = SATELLITE.IMU.gyro
+                elif SATELLITE.IMU_NAME == "BMX160":
+                    accel = SATELLITE.IMU.accel()
+                    mag = SATELLITE.IMU.mag()
+                    gyro = SATELLITE.IMU.gyro()
```

Comment:
This should not existm, the application layer should be independent of any specific IMU model. YOu have the match the fct interface @perrintong325